test/jasmine/tests/ternary_test.js

```diff
@@ -514,6 +514,33 @@ describe('ternary plots', function() {
         .then(done);
     });
 
+    describe('plotly_relayouting', function() {
+        ['pan', 'zoom'].forEach(function(dragmode) {
+            it('should emit plotly_relayouting events on ' + dragmode, function(done) {
+                var events = []; var path = [[350, 250], [375, 250], [375, 225]]; var relayoutCallback;
+                var fig = Lib.extendDeep({}, require('@mocks/ternary_simple'));
+                fig.layout.dragmode = dragmode;
+
+                var gd = createGraphDiv();
+                Plotly.plot(gd, fig)
+                .then(function() {
+                    relayoutCallback = jasmine.createSpy('relayoutCallback');
+                    gd.on('plotly_relayout', relayoutCallback);
+                    gd.on('plotly_relayouting', function(e) {
+                        events.push(e);
+                    });
+                    return drag(path);
+                })
+                .then(function() {
+                    expect(events.length).toEqual(path.length - 1);
+                    // expect(relayoutCallback).toHaveBeenCalledTimes(1);
```

Comment:
:zap:

Reply:
... you're right, we don't really need that `expect` statement. There's no other way to fill `events`.

Reply:
Oh wait, I see what's going on. `plotly_relayout` isn't emitted on `pan` as `ternary_simple` in zoomed out on first render.

Try 

```js
var fig = Lib.extendDeep({}, require('@mocks/ternary_simple'));
fig.layout.ternary.aaxis.min = 0.15
```

and then both `pan` and `zoom` should trigger a `plotly_relayout` on drag-end.

Reply:
I thought this was a bug last night but it's not. Thank you for figuring it out for me :male_detective:

Reply:
I wrote my last comment too fast. `ternary_simple` is not zoomed in on the first render and I am not sure why my `plotly_relayout` event listener is called :thinking:

Reply:
> I am not sure why my `plotly_relayout` event listener is called

You're calling `drag` here, so shouldn't `plotly_relayout` get called once at the end of the drag motion?

Reply:
It should but it doesn't at the moment... It's a bit disappointing because I would like to check that all the keys in the event data of `plotly_relayout` are also present in the event data of `plotly_relayouting`...

Reply:
Ok well no need to waste time on this detail. Resolved.